consent/strategy_default.go

```diff
@@ -745,10 +745,42 @@ func (s *DefaultStrategy) issueLogoutVerifier(w http.ResponseWriter, r *http.Req
 	// logout flow.
 	redir := s.c.LogoutRedirectURL().String()
 
-	// The hint must be set if it's an RP-initiated logout flow.
-	hint := r.URL.Query().Get("id_token_hint")
-	state := r.URL.Query().Get("state")
-	requestedRedir := r.URL.Query().Get("post_logout_redirect_uri")
+	var hint string
+	var state string
+	var requestedRedir string
+
+	if r.Method == http.MethodGet {
```

Comment:
I think you should just do the same as how [form processing is done in fosite](https://github.com/ory/fosite/blob/master/authorize_request_handler.go#L224-L226):

```
	if err := r.ParseMultipartForm(1 << 20); err != nil && err != http.ErrNotMultipart {
		return request, errors.WithStack(ErrInvalidRequest.WithHint("Unable to parse HTTP body, make sure to send a properly formatted form request body.").WithDebug(err.Error()))
	}
```

And then just access `r.Form`. Go already knows how to parse either query string or post payload. This is how it is done in fosite anyway. No need to be too precise here I think.